### ISSUE TITLE
truncate for rings with negative degrees

### ISSUE THREAD

**Issue by mahrud** [Macaulay2/M2#2001]
@mikestillman is the method `truncate` correct for multigraded rings with negative degrees? The documentation page for [Truncations](https://faculty.math.illinois.edu/Macaulay2/doc/Macaulay2/share/doc/Macaulay2/Truncations/html/index.html) doesn't sound too confident:
> This package handles the multi-graded case correctly, at least for the case of non-negative degree vectors for the variables, and the truncate function is functorial (see truncate(List,Matrix)).

Is there any place to read about what algorithm it is using? I see that it involves polyhedra, but it's not clear what it is doing.

I'm interested in something like the Hirzebruch surface. The computation below seems to be correct:

```m2
i1 : needsPackage "NormalToricVarieties"

o1 = NormalToricVarieties

o1 : Package

i2 : needsPackage "Truncations"

o2 = Truncations

o2 : Package

i3 : X = hirzebruchSurface(2)

o3 = X

o3 : NormalToricVariety

i4 : S = ring X

o4 = S

o4 : PolynomialRing

i5 : truncate({-2,1}, S^{{0,0}})

o5 = image | x_1 x_3 |

                             1
o5 : S-module, submodule of S

i6 : basis({-2,1}, S^{{0,0}})

o6 = | x_1 |

             1       1
o6 : Matrix S  <--- S

i7 : basis({0,1}, S^{{0,0}})

o7 = | x_0^2x_1 x_0x_1x_2 x_1x_2^2 x_3 |

             1       4
o7 : Matrix S  <--- S
```

**Comment by mahrud** [comment]
@mikestillman I think we figured out the issues in truncate. It's not correct for the Hirzebruch surface. Fixing it in this case is not too hard, but I'm not sure how to fix it for cases when the nef cone is not the positive quadrant.

Also, mentioning this bug here so I don't forget to fix it:
```m2
R = QQ[x,y, DegreeRank => 2];
truncate({1,1}, R^0) -- stdio:7:1:(3): error: expected more than 0 arguments
```

**Comment by mahrud** [comment]
Fixed in #2289.
